packages/validator/contracts/ERC20Executor.sol

```diff
@@ -0,0 +1,72 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import "./SingleClaimantExecutor.sol";
+import "@openzeppelin/contracts/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * @dev An abstract implementation of an Executor that only allows claims that have a higher nonce than previously seen to execute.
+ *      To use, subclass and override `executeClaim` and `metadata`, being sure to call `super` inside `executeClaim` before doing anything else.
+ *      This executor expects the first 32 bytes of `claimData` to be the nonce; you may optionally use extra data for your own purposes.
+ */
+abstract contract ERC20Executor is SingleClaimantExecutor {
+    event ClaimedERC20(address issuer, address from, address beneficiary, address token, uint256 amount, uint256 expiration);
+    
+    error ClaimCodeExpired();
+
+    constructor(address _validator) SingleClaimantExecutor(_validator) { }
+  
+    /**
+     * @dev Executes a claim that has been verified by the `ValidatorRegistry`. Implementers must check that this function
+     *      was called by a registry they recognise, and that any conditions in claimData such as replay protection are met
+     *      before acting on the request.
+     * @param issuer The account that issued the claim.
+     * @param claimant The account that is entitled to make the claim.
+     * @param beneficiary The account that the claim should benefit.
+     * @param claimData Claim data provided by the issuer.
+     * @param executorData Contextual information stored on the ValidatorRegistry for this issuer.
+     */
+    function executeClaim(address issuer, address claimant, address beneficiary, bytes calldata claimData, bytes calldata executorData) public virtual override {
+        super.executeClaim(issuer, claimant, beneficiary, claimData, executorData);
+        (
+         address from,
```

Comment:
Do you have a specific use for this in mind? If so I'm happy to merge it, but otherwise I'd rather make the 'standard' executors compatible with non-configurable issuers, meaning they should expect at most a nonce from the issuer, and all this information would be configured by the owner of the issuer when they call `claimAndConfigure` on the `ValidatorRegistry`.

Reply:
I had in mind the use case of generating claim links/QR-codes on the fly such that different claims could have different token amounts. I didn't get it initially, but now I see the limitation of non-configurable issuers and I can rewrite it as you suggested. 

I'm also thinking about the possible use-case of dapps/wallets supporting gift links in form of claim links/QR-codes. E.g. user creating gift link for an NFT. In this scenario, it would make sense to set up executor and configure validator per user only once and pass the NFT details in claim data, which might differ per each gift link. Can it be a part of the Shibboleth protocol? Or is it better to leave Shibboleth for physical devices only?

Reply:
I think that's definitely a good use-case. For issuers that are configurable, though, it might make sense to just write a custom validator - since you can enter the validator address into the issuer directly - and save the ValidatorRegistry for use with issuers that don't have any built-in configurability.

Reply:
Oh, got it. Makes sense. Ok, I will rewrite this Executor for non-configurable issuers

Reply:
@Arachnid Updated: Claimed data moved to Executor's `configdata`. That should allow non-configurable devices. Could you please take a look?